backend/utils/threads_utils.py

```diff
@@ -28,6 +28,39 @@ async def get_message_threads(org_id, room_id, message_id):
     message = await get_message(org_id, room_id, message_id)
     return message["threads"]
 
+async def get_message_thread(org_id, room_id, message_id, thread_id):
+    """Retrives a single the messages in a thread.
+
+    Args:
+        org_id (str): The organization id where the message is being updated.
+        room_id (str): The id of the room the message was sent in.
+        message_id (str): The id of the parent message whose thead message is to be edited.
+        thread_id (str): The id of the thread to be edited.
+
+
+    Returns:
+        [dict]: Returns an array of thread message objects.
+    """
+
+    query = {
+        "_id": message_id,
+        "threads.thread_id": thread_id
+    }
+
+    options = { "projection": {"threads.$": 1} }
+
+    response = await DataStorage(org_id).read(
+        collection_name=settings.MESSAGE_COLLECTION,
+        query=query,
+        options=options
+    )
+    
+    if not response or "status_code" in response:
+        return []
+
+    return response["threads"]
+
```

Comment:
I am struggling to understand what this function does exactly.
Does it retrieve a single thread from a list of a message's threads or all the threads under that message?

I ask because, if it is a single thread message, then this should probably return an object (dictionary) instead of a list.
will appreciate a little bit of clarification

Reply:
Yes, it retrieves a single thread from the list of threads in a message. But because the `threads` field in the message object is a **list** of thread objects, a query to zc_core would return a **list** as well (same data type) of a single thread (by thread_id specified). 

But as per your suggestion to make the function return an object instead, I could change it to return `response["threads"][0]` .